src/lib.rs

```diff
@@ -447,7 +472,22 @@ impl Client {
     /// Used internally to modify the options map of an _already cloned_
     /// [`Client`] instance.
     pub(crate) fn add_option(&mut self, name: impl Into<String>, value: impl Into<String>) {
-        self.options.insert(name.into(), value.into());
+        self.options.insert(Cow::Owned(name.into()), value.into());
+    }
+
+    pub(crate) fn set_role(&mut self, role: Option<String>) {
+        // By setting the role via `options`, we can be sure we've overwritten any role
+        // manually set by the user with `with_option()`.
```

Comment:
I thought this might be something we want, since it is technically already possible to set roles via `with_option()`. Otherwise the user would have to go through their code to check for any existing `with_option("role", ...)` invocations and replace it for this to work properly (assuming they even _realized_ that's something they needed to do).

Reply:
This is an internal method called from the others.